sqldb/sqlc/migrations/000001_invoices.up.sql

```diff
@@ -0,0 +1,152 @@
+-- invoices table contains all the information shared by all the invoice types. 
+CREATE TABLE IF NOT EXISTS invoices (
+    -- The id of the invoice. Translates to the AddIndex.
+    id INTEGER PRIMARY KEY,
+
+    -- The hash for this invoice. The invoice hash will always identify that 
+    -- invoice.
+    hash BLOB NOT NULL UNIQUE,
+
+    -- The preimage for the hash in this invoice. Some invoices may have this 
+    -- field empty, like unsettled hodl invoices or AMP invoices.
+    preimage BLOB,
+
+    -- An optional memo to attach along with the invoice. 
+    memo TEXT,
+
+    -- The amount of the invoice in millisatoshis.
+    amount_msat BIGINT NOT NULL, 
+
+    -- Delta to use for the time-lock of the CLTV extended to the final hop.
+    cltv_delta INTEGER NOT NULL,
+
+    -- The time before this invoice expiries, in seconds.
+    expiry INTEGER NOT NULL,
+
+    -- A randomly generated value include in the MPP record by the sender to 
+    -- prevent probing of the receiver.
+    payment_addr BLOB UNIQUE,
+
+    -- The encoded payment request for this invoice. Some invoice types may 
+    -- not have leave this empty, like Keysends.
+    payment_request TEXT UNIQUE, 
+
+    -- The invoice state.
+    state SMALLINT NOT NULL,
+
+    -- The accomulated value of all the htlcs settled for this invoice.
+    amount_paid_msat BIGINT NOT NULL,
+
+    -- This field will be true for AMP invoices.
+    is_amp BOOLEAN NOT NULL,
+
+    -- This field will be true for hodl invoices, independently of they being
+    -- settled or not.
+    is_hodl BOOLEAN NOT NULL,
+
+    -- This field will be true for keysend invoices.
+    is_keysend BOOLEAN NOT NULL,
+
+    -- Timestamp of when this invoice was created.
+    created_at TIMESTAMP NOT NULL
+);
+
+CREATE INDEX IF NOT EXISTS invoices_hash_idx ON invoices(hash);
+CREATE INDEX IF NOT EXISTS invoices_preimage_idx ON invoices(preimage);
+CREATE INDEX IF NOT EXISTS invoices_payment_addr_idx ON invoices(payment_addr);
+CREATE INDEX IF NOT EXISTS invoices_state_idx ON invoices(state);
+CREATE INDEX IF NOT EXISTS invoices_created_at_idx ON invoices(created_at);
+
+-- invoice_features contains the feature bits of an invoice.
+CREATE TABLE IF NOT EXISTS invoice_features (
+    -- The feature bit.
+    feature INTEGER NOT NULL,
+
+    -- The invoice id this feature belongs to.
+    invoice_id INTEGER NOT NULL REFERENCES invoices(id),
+
+    -- The feature bit is unique per invoice.
+    UNIQUE (feature, invoice_id)
+);
+
+CREATE INDEX IF NOT EXISTS invoice_feature_invoice_id_idx ON invoice_features(invoice_id);
+
+-- invoice_htlcs contains the information of a htlcs related to one of the node 
+-- invocies. 
+CREATE TABLE IF NOT EXISTS invoice_htlcs (
+    -- The id for this htlc. Used in foreign keys instead of the 
+    -- htlc_id/chan_id combination.
+    id INTEGER PRIMARY KEY,
+
+    -- The uint64 htlc id stored as text.
+    htlc_id TEXT NOT NULL,
```

Comment:
Exactly, the problem is that we do not have uint64, another option could be to store it as 8 bytes, but it could be more tedious for other apps or users to use it in the queries.

Reply:
Discussed this a bit, but I think we could just use a signed 8 byte integer here?

Reply:
the main reason was that set_id is 32 bytes vs htlc_id which is a uint64. 

We do not have an easy way to support uint64 in postgres/sqlite (it is not a type in the SQL standard), but given that htlc_id MUST BE a counter, I think it is safe to store it as int64 and just check that we do not get out of range 

Thanks @bhandras for the tip :+1: